compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java

```diff
@@ -82,10 +92,18 @@ public Solution recover(ParserRuleContext currentCtx, STToken nextToken, Object.
             return fix;
         }
 
+        int currentIndex = this.tokenReader.getCurrentIndex();
+        if (currentIndex == this.previousIndex) {
+            itterCount++;
+        } else {
+            itterCount = 0;
+            previousIndex = currentIndex;
+        }
+
         Result bestMatch = seekMatch(currentCtx);
         if (bestMatch.matches > 0) {
             Solution sol = bestMatch.solution;
-            if (sol != null) {
+            if (sol != null && itterCount < ITTER_LIMIT) {
```

Comment:
Can we move this check to before calling `seekMatch()`?

Reply:
okay